File: assignment 4/python/program1.py
class Time:
    def __init__(self, hours, minutes, seconds):
        self.hours = hours
        self.minutes = minutes
        self.seconds = seconds

    def to_seconds(self):
        return self.hours * 3600 + self.minutes * 60 + self.seconds

def get_time_input(label):
    h, m, s = map(int, input(f"Enter {label} time (hh mm ss): ").split())
    return Time(h, m, s)

def main():
    start = get_time_input("start")
    end = get_time_input("end")

    start_seconds = start.to_seconds()
    end_seconds = end.to_seconds()

    diff_seconds = end_seconds - start_seconds

    # Handle wrap-around if event ends the next day
    if diff_seconds < 0:
        diff_seconds += 24 * 3600

    # Convert back to hh:mm:ss
    hours = diff_seconds // 3600
    diff_seconds %= 3600
    minutes = diff_seconds // 60
    seconds = diff_seconds % 60

    print(f"Difference: {hours} hours, {minutes} minutes, {seconds} seconds")
    print(f"Total difference in seconds: {hours * 3600 + minutes * 60 + seconds}")

if __name__ == "__main__":
    main()

File: assignment 4/C programming/program1.c
//Write a program in C to calculate difference between start time and end time of an event. Time is expressed as hr-min-sec. Final result to be produced in seconds
#include <stdio.h>

struct Time {
    int hours;
    int minutes;
    int seconds;
};

int main() {
    struct Time start, end, diff;

    // Input start time
    printf("Enter start time (hh mm ss): ");
    scanf("%d %d %d", &start.hours, &start.minutes, &start.seconds);

    // Input end time
    printf("Enter end time (hh mm ss): ");
    scanf("%d %d %d", &end.hours, &end.minutes, &end.seconds);

    // Convert start time and end time to seconds
    int startInSeconds = start.hours * 3600 + start.minutes * 60 + start.seconds;
    int endInSeconds = end.hours * 3600 + end.minutes * 60 + end.seconds;

    // Calculate difference in seconds
    int diffInSeconds = endInSeconds - startInSeconds;

    // If the difference is negative, it means the event ended the next day
    if (diffInSeconds < 0) {
        diffInSeconds += 24 * 3600;  // Add 24 hours in seconds
    }

    // Convert difference back to hours, minutes, and seconds
    diff.hours = diffInSeconds / 3600;
    diffInSeconds %= 3600;
    diff.minutes = diffInSeconds / 60;
    diff.seconds = diffInSeconds % 60;

    // Output the result
    printf("Difference: %d hours, %d minutes, %d seconds\n", diff.hours, diff.minutes, diff.seconds);
    printf("Total difference in seconds: %d\n", (diff.hours * 3600 + diff.minutes * 60 + diff.seconds));

    return 0;
}